Automattic/jetpack
# VideoPress: Expose playback token generation endpoint

Relates to #26675. This is part of the solution for the private video thumbnail issue. See comments on the original issue.

#### Changes proposed in this Pull Request:

* Add a new endpoint that uses the WPCOM API to generate a playback token for a given video GUID

#### Other information:


#### Jetpack product discussion



N/A

#### Does this pull request change what data or activity we track or use?



No.

#### Testing instructions:






* If you don't have videos, upload at least one to the VideoPress library
* Get the `GUID` for the uploaded video (see how to do it below)
* Do a `POST` request to the new `/wpcom/v2/videopress/playback-jwt/{ GUID }` endpoint:

```
curl -X POST -u "user:pass" http://yoursite.com/wp-json/wpcom/v2/videopress/playback-jwt/{ GUID } | jq
```
* You will receive a response like this:

```
{
    "playback_token": "eAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA9.eyJpc3MiAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA30.6vUAAAAAAAAAAAAAAAAAo3nE"
}
```

**How to get a video GUID:**

* click in the `Edit video details` button
* copy the video URL, for example:

```
https://videos.files.wordpress.com/K0BBgf00/sample-file.mp4
```

* The GUID is the portion of the URL right after `https://videos.files.wordpress.com/` and before the file name portion
* In the example, the GUID is `K0BBgf00`
